cassandra/src/main/scala/akka/stream/contrib/cassandra/CassandraSource.scala

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib.cassandra
+
+import akka.NotUsed
+import akka.stream._
+import akka.stream.scaladsl.Source
+import akka.stream.stage.{ OutHandler, GraphStageLogic, GraphStage }
+import com.datastax.driver.core.{ ResultSet, Session, Statement, Row }
+import com.google.common.util.concurrent.{ ListenableFuture, FutureCallback, Futures }
+import scala.collection.mutable
+import scala.concurrent.{ Future, Promise }
+import scala.collection.JavaConverters._
+import scala.util.{ Try, Failure, Success }
+
+class CassandraSource(stmt: Statement, session: Session) extends GraphStage[SourceShape[Row]] {
```

Comment:
When I implemented the other `SelectSource` I found it useful to have the `Statement` param as `Future[Statement]`. That makes it easier to compose with other async things like `prepareAsync`.